### PR TITLE
Test empty named argument serialization

### LINKED ISSUE #1030: Empty array sent instead of empty object when named arguments should be used
We should add tests for sending requests with named arguments when the map of arguments is empty.

_Originally posted by @prodigy in https://github.com/microsoft/vs-streamjsonrpc/discussions/1029#discussioncomment-9073068_

### PR DESCRIPTION
**Pull request by AArnott**
Issue #1030 requested coverage to ensure parameter-object calls do not serialize empty arguments as `params: []`.

This adds wire-level tests for:
- omitted and explicitly null parameter objects, which continue to omit `params`
- null and empty named-argument dictionaries, ensuring they are omitted or serialized as `{}` rather than `[]`
- source-generated and dynamic proxies with `ServerRequiresNamedArguments`

No product change is required because the reported behavior does not reproduce on the current implementation.

Fixes #1030

### CODE REVIEW

File: test/StreamJsonRpc.Tests/JsonRpcProxyGenerationTests.cs [hunk 1]
```diff
@@ -415,6 +416,20 @@ public async Task CallMethod_void_void()
...
+        var messageHandler = new CapturingMessageHandler();
+        using JsonRpc clientRpc = new(messageHandler);
+        IServer proxy = clientRpc.Attach<IServer>(new JsonRpcProxyOptions(this.DefaultProxyOptions) { ServerRequiresNamedArguments = true });
+        clientRpc.StartListening();
+
+        _ = proxy.IncrementAsync();
+        JToken request = await messageHandler.WrittenMessages.DequeueAsync(this.TimeoutToken);
+
+        Assert.NotEqual(JTokenType.Array, request["params"]?.Type);
+    }
```
[the remark this answers is not in the document]
Comment (AArnott) [reply test/StreamJsonRpc.Tests/JsonRpcProxyGenerationTests.cs]: Changed the regression test to use a parameterless void notification proxy, so it exercises the same named-argument serialization without leaving a response-bearing invocation pending. The assertion now explicitly accepts only omitted `params` or a JSON object.

> [!NOTE]
> Auto-replied by the [GitHub Copilot app](https://github.com/github/app).

**Review by RyanToth3 — APPROVED** [review]
